extract-stream: use subclass instead of one-off transformer

isaacs opened the pull request:
This addresses the root cause of why duplicate end events keep happening.

isaacs committed:
extract-stream: use subclass instead of one-off transformer

This addresses the root cause of why duplicate end events keep happening.

zkat reviewed (commented):
Does this sort of thing have issues in node@6? Getting weird timeouts for it. I'm running the tests now to see if it was a fluke.

isaacs committed:
test: don't use gpg signing for test commit

Makes tests noisy and interactive if the user's GPG key requires
passphrase or tfa.

isaacs committed:
tests: listen for data before listening to end
